### ISSUE TITLE
Feature request : format Value before apply validation

### ISSUE THREAD

**Issue by mseld** [trenskow/isvalid#19]
## Formatter

It would be nice to have a formatter for any property like upper-case , lower-case or custom format

```javascript
{
    type: String,
    formater : 'lower-case'
    enum: ["day", "month", "year"]
}
```
or something like that
```javascript
{
    type: String,
    formater: function(data, schema) {
           return data.toLowerCase();
     },
    enum: ["day", "month", "year"]
}
```

**Comment by trenskow** [comment]
I agree. I would, though, call it `transform` instead, and make it a generic validator for all types. I will look into it.

**Comment by trenskow** [comment]
After thinking a little more about this, I actually think the `custom` validator should become a hook. I think I'll rethink it as something like this.

    {
        type: String,
        on: {
            pre: function(data, schema) {
                // Pre validation
            },
            post: function(data, schema) {
                // Post validation
            }
        }
    }
